pkg/server/server.go

```diff
@@ -339,6 +341,27 @@ func (s *Server) Run(ctx context.Context) error {
 		return err
 	}
 
+	if s.Options.Cache.Enabled {
+		if err := s.AddPostStartHook("kcp-start-optional-informers", func(hookContext genericapiserver.PostStartHookContext) error {
+			// TODO(p0lyn0mial): failing the optional hook should not render the main server unhealthy
```

Comment:
thoughts?

Reply:
why can this fail?

Reply:
the cache server is an external process, we might fail on synchronizing informers